smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/endpoints/EndpointParametersGenerator.kt

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.smithy.kotlin.codegen.rendering.endpoints
+
+import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.kotlin.codegen.core.KotlinWriter
+import software.amazon.smithy.kotlin.codegen.core.withBlock
+import software.amazon.smithy.kotlin.codegen.lang.KotlinTypes
+import software.amazon.smithy.kotlin.codegen.model.boxed
+import software.amazon.smithy.kotlin.codegen.utils.doubleQuote
+import software.amazon.smithy.kotlin.codegen.utils.getOrNull
+import software.amazon.smithy.rulesengine.language.EndpointRuleset
+import software.amazon.smithy.rulesengine.language.eval.Value
+import software.amazon.smithy.rulesengine.language.lang.Identifier
+import software.amazon.smithy.rulesengine.language.lang.parameters.Parameter
+import software.amazon.smithy.rulesengine.language.lang.parameters.ParameterType
+
+private const val DEFAULT_DEPRECATED_MESSAGE =
+    "This field is deprecated and no longer recommended for use."
+
+private const val CLASS_NAME = "EndpointParameters"
+
+/**
+ * Renders the struct of parameters to be passed to the endpoint provider for resolution.
+ */
+class EndpointParametersGenerator(
+    private val writer: KotlinWriter,
+    rules: EndpointRuleset,
+) {
+    private val params: List<KotlinEndpointParameter> = rules.parameters.toList().map {
+        KotlinEndpointParameter(
+            it.name.toKotlin(),
+            it.type.toSymbol(),
+            it.isRequired,
+            it.defaultValue.getOrNull()?.toKotlinLiteral() ?: "null",
+            it.documentation.getOrNull(),
```

Comment:
Question: Do we run the ruleset documentation through our sanitization? Is that just free because it's part of the model or do we need to explicitly handle that?